app/controllers/submissions_controller.rb

```diff
@@ -75,4 +27,17 @@ def submission_params
   def set_submission
     @submission = Submission.by_user(current_user).find(params[:id])
   end
+
+  def handle_successful_update(format)
+    format.html do
+      flash[:success] = I18n.t("submission_updated")
+      redirect_to submission_path(@submission.phase)
```

Comment:
this should redirect to the submission id, not the phase id
```suggestion
      redirect_to submission_path(@submission)
```